File: src/2544.alternating-digit-sum.2630/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var alternateDigitSum = function (n) {
    let result = 0;
    let sign = 1;
    const str = n.toString();
    for (let i = 0; i < str.length; i++) {
        result += sign * (+str[i]);
        sign = -sign;
    }
    return result;
};

File: src/2544.alternating-digit-sum.2630/solution.java
class Solution {
    public int alternateDigitSum(int n) {
        int sign = String.valueOf(n).length()%2 == 1?1:-1;
        int result = 0;
        while( n != 0){
            result += sign*(n%10);
            n /= 10;
            sign *= -1;
        }
        return result;
    }
}